test/widgets/app_test.dart

```diff
@@ -37,24 +37,53 @@ void main() {
     }
 
     testWidgets('when no accounts, go to choose account', (tester) async {
+      check(testBinding.globalStore).accounts.isEmpty();
+      check(testBinding.globalStore).lastVisitedAccount.isNull();
       await prepare(tester);
       check(pushedRoutes).deepEquals(<Condition<Object?>>[
         (it) => it.isA<WidgetRoute>().page.isA<ChooseAccountPage>(),
       ]);
     });
 
-    testWidgets('when have accounts, go to home page for first account', (tester) async {
-      // We'll need per-account data for the account that a page will be opened
-      // for, but not for the other account.
-      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
-      await testBinding.globalStore.insertAccount(eg.otherAccount.toCompanion(false));
-      await prepare(tester);
+    group('when have accounts', () {
+      testWidgets('with account(s) visited, go to home page for the last visited account', (tester) async {
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+        await testBinding.globalStore.add(eg.otherAccount, eg.initialSnapshot(
+          realmUsers: [eg.otherUser]));
+        check(testBinding.globalStore).lastVisitedAccount.equals(eg.otherAccount);
+        await prepare(tester);
+
+        check(pushedRoutes).deepEquals(<Condition<Object?>>[
+          (it) => it.isA<MaterialAccountWidgetRoute>()
+            ..accountId.equals(eg.otherAccount.id)
+            ..page.isA<HomePage>(),
+        ]);
+      });
 
-      check(pushedRoutes).deepEquals(<Condition<Object?>>[
-        (it) => it.isA<MaterialAccountWidgetRoute>()
-          ..accountId.equals(eg.selfAccount.id)
-          ..page.isA<HomePage>(),
-      ]);
+      // testWidgets('when just upgraded to get the last-visited-account feature, '
+      //     'if there is at least one account, open that one, not the choose-account page', (tester) async {
+      //   // The database migration step from9To10 ensures that
+      //   // [IntGlobalSetting.lastVisitedAccountId], thus [GlobalStore.lastVisitedAccount],
+      //   // points to the first account in the list. That migration step has
+      //   // its own tests in test/model/database_test.dart.
+      // });
```

Comment:
In general we don't check in commented-out code.

The comment within the body here would be fine, with a bit of editing to replace the context given by the (commented-out) test name. But I think it isn't necessary: from this code's perspective, there just either is or isn't a last-visited account, so the two test cases above and below here cover those cases.

(I'm guessing this test was originally written for a previous revision of this PR where this behavior was implemented in `widgets/app.dart` rather than in the DB migration. For readers of the PR thread, it could be useful to explain why this test isn't needed, because they saw the stage where it would have been needed. But readers of this code or this commit don't have that history, and as a result don't need that explanation.)